models/sessions/scratch/bigquery/snowplow_web_sessions_this_run.sql

```diff
@@ -54,13 +54,37 @@ with session_firsts as (
         mkt_campaign as mkt_campaign,
         mkt_clickid as mkt_clickid,
         mkt_network as mkt_network,
+        regexp_extract(page_url ,r'utm_source_platform=([^?&#]*)') as mkt_source_platform,
+        case
+            when mkt_source = '(direct)' and mkt_medium in ('(not set)', '(none)') then 'Direct'
+            when CONTAINS_SUBSTR(mkt_medium, 'cross-network') then 'Cross-network'
+            when REGEXP_CONTAINS(mkt_medium, r'^(.*cp.*|ppc|retargeting|paid.*)$') then
+            case
+                when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SHOPPING')}}) or REGEXP_CONTAINS(mkt_campaign, r'^(.*(([^a-df-z]|^)shop|shopping).*)$)') then 'Paid Shopping'
+                when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SEARCH')}}) then 'Paid Search'
+                when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SOCIAL')}}) then 'Paid Social'
+                when mkt_source in ({{select_source_category('SOURCE_CATEGORY_VIDEO')}}) then 'Paid Video'
+                else 'Paid Other'
+            end
+            when mkt_medium in ('display', 'banner', 'expandable', 'intersitial', 'cpm') then 'Display'
+            when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SHOPPING')}}) or REGEXP_CONTAINS(mkt_campaign, r'^(.*(([^a-df-z]|^)shop|shopping).*)$') then 'Organic Shopping'
+            when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SOCIAL')}}) or mkt_medium in ('social', 'social-network', 'sm', 'social network', 'social media') then 'Organic Social'
+            when mkt_source in ({{select_source_category('SOURCE_CATEGORY_VIDEO')}}) or REGEXP_CONTAINS(mkt_medium, r'^(.*video.*)$') then 'Organic Video'
+            when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SEARCH')}}) or mkt_medium = 'organic' then 'Organic Search'
+            when mkt_medium in ('referral', 'app', 'link') then 'Referral'
+            when mkt_source in ('email', 'e-mail', 'e_mail', 'e mail') OR mkt_medium in ('email', 'e-mail', 'e_mail', 'e mail') then 'Email'
+            when mkt_medium = 'affiliate' then 'Affiliates'
+            when mkt_medium = 'audio' then 'Audio'
+            when mkt_source = 'sms' or mkt_medium = 'sms' then 'SMS'
+            when ENDS_WITH(mkt_medium, 'push') or REGEXP_CONTAINS(mkt_medium, r'.*(mobile|notification).*') or mkt_source = 'firebase' then 'Mobile Push Notifications'
+        end as default_channel
```

Comment:
Dumb question for me, why is it called default channel?

Reply:
Also, missing comma at end of the line

Reply:
We are recreating [this](https://support.google.com/analytics/answer/9756891?hl=en) GA4 dimension here, and I realized it should be named `default_group_channel`.
Will update the name!